frontend/packages/dev-console/src/components/import/image-search/ImageStream.tsx

```diff
@@ -0,0 +1,192 @@
+      acc[namespace] = namespace;
+    }
+    return acc;
+  }, projectList);
+
+  const [imageNamespace] = useField('imageStream.namespace');
+  const [imageTag] = useField('imageStream.tag');
+  const [imageName] = useField('imageStream.image');
+  const { values, setFieldValue, setFieldError, initialValues } = useFormikContext<FormikValues>();
+  const [selectedProject, setselectedProject] = React.useState(imageNamespace.value);
+  const [selectedTag, setselectedTag] = React.useState(imageTag.value);
+  const [selectedName, setselectedName] = React.useState(imageName.value);
+
+  const searchImageTag = React.useCallback(() => {
+    setFieldValue('isSearchingForImage', true);
+    k8sGet(ImageStreamTagModel, `${imageName.value}:${imageTag.value}`, imageNamespace.value)
+      .then((imageStreamImport) => {
+        const { image, tag, status } = imageStreamImport;
+        const name = imageName.value;
+        const isi = { name, image, tag, status };
+        const ports = getPorts(isi);
+        setFieldValue('isSearchingForImage', false);
+        setFieldValue('isi.name', name);
+        setFieldValue('isi.image', image);
+        setFieldValue('isi.tag', imageTag.value);
+        setFieldValue('isi.ports', ports);
+        setFieldValue('image.ports', ports);
+        setFieldValue('name', getSuggestedName(name));
+        !values.application.name &&
+          setFieldValue('application.name', `${getSuggestedName(name)}-app`);
+        // set default port value
+        const targetPort = _.head(ports);
+        targetPort && setFieldValue('route.targetPort', makePortName(targetPort));
+      })
+      .catch((error) => {
+        setFieldError('isi.image', error.message);
+        setFieldValue('isi', {});
+        setFieldValue('isSearchingForImage', false);
+      });
+  }, [
+    setFieldValue,
+    imageName.value,
+    imageTag.value,
+    imageNamespace.value,
+    values.application.name,
+    setFieldError,
+  ]);
+  const getImageStreamByNamespace = (ns: string) => {
+    const imageStreamsList = {};
+    const isBuilderImageNamespace = ns === builderImagesNamespace.Openshift;
+    const imgStreams = isBuilderImageNamespace
+      ? (normalizeBuilderImages(imageStreams) as NormalizedBuilderImages)
+      : (imageStreams as K8sResourceKind[]);
+    _.each(imgStreams, (img: BuilderImage | K8sResourceKind) => {
+      const { name, namespace } = isBuilderImageNamespace
+        ? (img as BuilderImage).obj.metadata
+        : (img as K8sResourceKind).metadata;
+      if (namespace === ns) {
+        imageStreamsList[name] = name;
+      }
+    });
+    return imageStreamsList;
+  };
+  const getImageStreamTags = (name: string) => {
+    const tags = {};
+    const imageStream = _.find(imageStreams, ['metadata.name', name]);
+    name &&
+      !_.isEmpty(imageStream) &&
+      _.each(imageStream.status.tags, ({ tag }) => {
+        tags[tag] = tag;
+      });
+    return tags;
+  };
+  React.useEffect(() => {
```

Comment:
Again i think this is not necessary here. You can just use `values.something` along with `useEffect` hook to reset your form values.